pkg/process/v6/writer.go

```diff
@@ -96,6 +102,79 @@ func (w writer) writeEntry(entry transformers.RelatedEntries) error {
 	return nil
 }
 
+func (w *writer) normalizeSeverity(handle *grypeDB.VulnerabilityHandle) {
```

Comment:
I think at the very least, a comment would help this one. As I understand it, this is what's happening:
* if an nvd record is processed, the top severity is tracked
* if a non-nvd record is processed without a severity, it's looked up from the nvd severity captured earlier

Is this accurate? `normalizeServerity` doesn't indicate what this behavior is especially well to me, maybe call it `trackNvdSeverityAndApplyToMatchingCVEsWithoutSeverity` is a terrible name, but something different here I think would help, at the very least a big comment explaining this.

And how are we guaranteed that NVD runs first?

Reply:
> Is this accurate?

it is! I'll rename and add some comments around this 👍 

> how are we guaranteed that NVD runs first?

We do already have code when iterating through provider configs (which is the same order we process provider results) to ensure that [NVD is always run first](https://github.com/anchore/grype-db/blob/da8aec0da8c9b38e9905e53e83150de7fce5b6fb/pkg/provider/providers/providers.go#L35-L37)